src/shared/color-utils.ts

```diff
@@ -284,101 +307,129 @@ function setAlpha(color: string, value: number): string {
     return recomposeColor(decomposedColor);
 }
 
-function createColorShades(color: string, darker = false): Record<string, string> {
+/**
+ * Calculates the contrast ratio between two colors.
+ *
+ * Formula: https://www.w3.org/TR/WCAG20-TECHS/G17.html#G17-tests
+ * @param {string} foreground - CSS color, i.e. one of: #nnn, #nnnnnn, rgb(), rgba(), hsl(), hsla()
+ * @param {string} background - CSS color, i.e. one of: #nnn, #nnnnnn, rgb(), rgba(), hsl(), hsla()
+ * @returns {number} A contrast ratio value in the range 0 - 21.
+ */
+function getContrastRatio(foreground: string, background: string): number {
```

Comment:
This is cool I like that we can programmatically ensure the colours are accessible, should we do this for every component though and only give limited options for the user? I know we talked about this already but I don't think there was a concrete decision yet.

Reply:
well, this is to be used in the theming creation part (apart from the default themes which ideally are hard-coded) ... so you are good, when you use `primary.contrast` for instance